src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
-
+enum FigureShape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum FigureColor {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: FigureShape;
+  color: FigureColor;
+  getArea: () => number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = FigureShape.Triangle;
+
+  constructor(
+    public color: FigureColor,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const isNotTraingle1 = a <= 0 || b <= 0 || c <= 0;
+    const isNotTraingle2 = a + b <= c || a + c <= b || b + c <= a;
+
+    if (isNotTraingle1 || isNotTraingle2) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
+  }
 
+  getArea = (): number => {
+    const s = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+      ) / 100
+    );
+  };
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = FigureShape.Circle;
+
+  constructor(
+    public color: FigureColor,
+    public circle: number,
+  ) {
+    if (circle < 0) {
```

Comment:
Radius?
```suggestion
    if (radius < 0) {
```

Reply:
)) yes, radius, i can't explane why i choose this word